src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs

```diff
@@ -103,13 +103,13 @@ public ValidationParameters()
         public ValidationParameters ActorValidationParameters { get; set; }
 
         /// <summary>
-        /// Gets or sets a delegate used to validate the cryptographic algorithm used.
+        /// Allows overriding the delegate used to validate the cryptographic algorithm used.
         /// </summary>
         /// <remarks>
-        /// If set, this delegate will validate the cryptographic algorithm used and
-        /// the algorithm will not be checked against <see cref="ValidAlgorithms"/>.
+        /// If no delegate is set, the default implementation will be used. The default checks the algorithm
+        /// against the <see cref="ValidAlgorithms"/> property, if present. If not, it will succeed.
         /// </remarks>
-        public AlgorithmValidator AlgorithmValidator { get; set; }
+        public AlgorithmValidationDelegate AlgorithmValidator { get; set; } = Validators.ValidateAlgorithm;
```

Comment:
Should we allow this to be set to null?

Reply:
I am still bouncing this idea in my head. Whether they should set a mostly empty validator to skip, or be allowed to simply set to null to skip the call altogether.

Reply:
I thought we were keeping the naming consistent with our current methods by just adding 'Delegate' as a postfix. i.e: ValidateAlgorithmDelegate?

Reply:
You are right. It should be `AlgorithmValidatorDelegate`.